src/repository.rs

```diff
@@ -47,9 +47,11 @@ impl Repository {
     }
 
     pub fn open_path(dirfd: impl AsFd, path: impl AsRef<Path>) -> Result<Repository> {
+        let path = path.as_ref();
+
         // O_PATH isn't enough because flock()
-        let repository = openat(dirfd, path.as_ref(), OFlags::RDONLY, Mode::empty())
-            .context("Cannot open composefs repository")?;
+        let repository = openat(dirfd, path, OFlags::RDONLY, Mode::empty())
+            .with_context(|| format!("Cannot open composefs repository at {}", path.display()))?;
```

Comment:
I actually dropped that recently because it doesn't make completely sense in context of the `dirfd` parameter and I was trying to slowly remove use of `anyhow`.  Admittedly, `dirfd` is basically always `CWD` currently...

My idea here was that the caller should add the context because it was more likely to know the full path.  Of course I never added that to any of the callers :)

Is there a usual rule for this sort of thing in Rust?

Reply:
In this case I'd push the error prefixing to the CLI, right?